## PAIR PROGRAMMING ESTADÍSTICA I
---

Buscar un dataset que os guste, por ejemplo de Kaggle o de donde queráis extraerlo. De esta forma vamos a estudiar los diferentes estadísiticos que contiene en dataset seleccionado Para todas los ejercicios de pair programming de Estadística I, II y III usaremos el dataframe que se tome para el desarrollo de este ejercicio de pair programming.

In [2]:
# Importamos las librerias

import numpy as np
from scipy import stats 
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
from pydataset import data


Los objetivos de hoy son:

**EJERCICIO 1**

Buscar un conjunto de datos a analizar:
- Se recomienda que el conjunto de datos a analizar tenga variables numéricas y categóricas. 
Ya que también se analizarán.

In [3]:
# Cargamos y abrimos el archivo

df = pd.read_csv('cost-of-living.csv')
df.head()


Unnamed: 0      city      country    x1     x2    x3    x4    x5    x6  \
0           0     Delhi        India  4.90  22.04  4.28  1.84  3.67  1.78   
1           1  Shanghai        China  5.59  40.51  5.59  1.12  4.19  3.96   
2           2   Jakarta    Indonesia  2.54  22.25  3.50  2.02  3.18  2.19   
3           3    Manila  Philippines  3.54  27.40  3.54  1.24  1.90  2.91   
4           4     Seoul  South Korea  7.16  52.77  6.03  3.02  4.52  3.86   

     x7  ...     x47      x48     x49      x50      x51       x52       x53  \
0  0.48  ...   36.26   223.87  133.38   596.16   325.82   2619.46   1068.90   
1  0.52  ...  121.19  1080.07  564.30  2972.57  1532.23  17333.09   9174.88   
2  0.59  ...   80.32   482.85  270.15  1117.69   584.37   2694.05   1269.44   
3  0.93  ...   61.82   559.52  281.78  1754.40   684.81   3536.04   2596.44   
4  1.46  ...  108.30   809.83  583.60  2621.05  1683.74  21847.94  10832.90   

       x54   x55  data_quality  
0   586.35  7.96             1  
1  1382.83  5.01             1  
2   483.19  9.15             1  
3   419.02  7.80             1  
4  2672.23  3.47             1  

[5 rows x 59 columns]

### Análisis exploratorio del df 'cost-of-living'

In [4]:
# Nombres de las columnas

df.columns


Index(['Unnamed: 0', 'city', 'country', 'x1', 'x2', 'x3', 'x4', 'x5', 'x6',
       'x7', 'x8', 'x9', 'x10', 'x11', 'x12', 'x13', 'x14', 'x15', 'x16',
       'x17', 'x18', 'x19', 'x20', 'x21', 'x22', 'x23', 'x24', 'x25', 'x26',
       'x27', 'x28', 'x29', 'x30', 'x31', 'x32', 'x33', 'x34', 'x35', 'x36',
       'x37', 'x38', 'x39', 'x40', 'x41', 'x42', 'x43', 'x44', 'x45', 'x46',
       'x47', 'x48', 'x49', 'x50', 'x51', 'x52', 'x53', 'x54', 'x55',
       'data_quality'],
      dtype='object')

In [5]:
# Renombramos las columnas

df.rename(columns = {"Unnamed: 0":"indice",'city': 'ciudad', 'country': 'pais', "x1":"comida_x1_rest", "x2":"comida_x2_rest", "x3":"menu_mcdonals", "x4":"cerveza_nac_rest_0.5cl", "x5":"cerveza_imp_rest_0.33cl", "x6":"capuccino_rest",
       "x7":"coke_rest_0.33cl", "x8": "agua_rest_0.33cl", "x9":"leche_1l", "x10":"barra_pan", "x11":"arroz_1kg", "x12":"huevos_12ud", "x13":"queso_1kg", "x14":"pollo_1kg", "x15":"ternera_1kg", 
       "x16":"manzanas_1kg", 'x17':"platanos_1kg", 'x18':"orange_1kg", 'x19':"tomates_1kg", 'x20':"patatas_1kg", 'x21':"cebollas_1kg", 'x22':"lechuga", 'x23':"agua_super_1.5l", 'x24':"botella_vino_super", 
       'x25':"cerveza_nac_super_0.5cl", 'x26':"cerveza_imp_super_0.33cl",'x27':"tabaco_paq_20ud", 'x28':"billete_transporte_pub", 'x29':"abono_transporte_pub", 'x30':"taxi_start", 'x31': "taxi_1km", 
       'x32':"taxi_1h", 'x33':"gasolina_1l", 'x34':"volkswagen_coche_nuevo", 'x35':"toyota_coche_nuevo", 'x36':"gastos_basicos_apartamento",'x37':"precio_movil_1min", 'x38':"internet_mensual", 
       'x39':"cuota_gym", 'x40':"alq_pista_tenis", 'x41':"entrada_cine", 'x42':"guarderia_mensual", 'x43':"colegio_anual", 'x44':"vaqueros_levis", 'x45':"vestido_verano", 'x46':"deportivas_nike",
       'x47':"zapatos_cuero_homb", 'x48':"apart_1hab_centro", 'x49':"apart_1hab_periferia", 'x50':"apart_3hab_centro", 'x51':"apart_3hab_periferia", 'x52':"m2_centro", 'x53':"m2_periferias", 
       'x54':"media_pond_salario_neto", 'x55':"interes_hipoteca"}, inplace = True)

In [6]:
df.columns

Index(['indice', 'ciudad', 'pais', 'comida_x1_rest', 'comida_x2_rest',
       'menu_mcdonals', 'cerveza_nac_rest_0.5cl', 'cerveza_imp_rest_0.33cl',
       'capuccino_rest', 'coke_rest_0.33cl', 'agua_rest_0.33cl', 'leche_1l',
       'barra_pan', 'arroz_1kg', 'huevos_12ud', 'queso_1kg', 'pollo_1kg',
       'ternera_1kg', 'manzanas_1kg', 'platanos_1kg', 'orange_1kg',
       'tomates_1kg', 'patatas_1kg', 'cebollas_1kg', 'lechuga',
       'agua_super_1.5l', 'botella_vino_super', 'cerveza_nac_super_0.5cl',
       'cerveza_imp_super_0.33cl', 'tabaco_paq_20ud', 'billete_transporte_pub',
       'abono_transporte_pub', 'taxi_start', 'taxi_1km', 'taxi_1h',
       'gasolina_1l', 'volkswagen_coche_nuevo', 'toyota_coche_nuevo',
       'gastos_basicos_apartamento', 'precio_movil_1min', 'internet_mensual',
       'cuota_gym', 'alq_pista_tenis', 'entrada_cine', 'guarderia_mensual',
       'colegio_anual', 'vaqueros_levis', 'vestido_verano', 'deportivas_nike',
       'zapatos_cuero_homb', 'apart_1hab_ce

In [7]:
# Para ver todas las columnas

pd.options.display.max_columns= None 

In [8]:
df.head()

indice    ciudad         pais  comida_x1_rest  comida_x2_rest  \
0       0     Delhi        India            4.90           22.04   
1       1  Shanghai        China            5.59           40.51   
2       2   Jakarta    Indonesia            2.54           22.25   
3       3    Manila  Philippines            3.54           27.40   
4       4     Seoul  South Korea            7.16           52.77   

   menu_mcdonals  cerveza_nac_rest_0.5cl  cerveza_imp_rest_0.33cl  \
0           4.28                    1.84                     3.67   
1           5.59                    1.12                     4.19   
2           3.50                    2.02                     3.18   
3           3.54                    1.24                     1.90   
4           6.03                    3.02                     4.52   

   capuccino_rest  coke_rest_0.33cl  agua_rest_0.33cl  leche_1l  barra_pan  \
0            1.78              0.48              0.19      0.73       0.51   
1            3.96              0.52              0.32      2.68       2.57   
2            2.19              0.59              0.27      1.28       1.21   
3            2.91              0.93              0.51      1.63       1.19   
4            3.86              1.46              0.78      2.13       2.78   

   arroz_1kg  huevos_12ud  queso_1kg  pollo_1kg  ternera_1kg  manzanas_1kg  \
0       0.88         1.01       4.32       3.79         5.57          1.80   
1       1.18         2.17      18.01       4.67        12.79          2.22   
2       0.83         1.70       6.66       3.46         8.32          2.80   
3       0.89         1.73       5.06       4.19         6.64          2.66   
4       3.66         4.02      11.13      10.62        40.41          6.79   

   platanos_1kg  orange_1kg  tomates_1kg  patatas_1kg  cebollas_1kg  lechuga  \
0          0.75        1.01         0.60         0.36          0.41     0.70   
1          1.55        2.11         1.47         0.82          1.02     0.82   
2          1.52        2.08         1.24         1.35          2.10     1.13   
3          1.42        2.95         1.42         1.76          1.75     1.36   
4          3.70        6.40         6.36         3.86          2.81     2.40   

   agua_super_1.5l  botella_vino_super  cerveza_nac_super_0.5cl  \
0             0.36                9.79                     1.53   
1             0.62               13.97                     0.91   
2             0.42               20.66                     2.06   
3             0.65                8.84                     1.15   
4             1.06               15.08                     2.10   

   cerveza_imp_super_0.33cl  tabaco_paq_20ud  billete_transporte_pub  \
0                      2.51             4.28                    0.49   
1                      1.93             3.49                    0.56   
2                      3.18             2.22                    0.22   
3                      1.55             2.65                    0.53   
4                      2.49             3.39                    0.98   

   abono_transporte_pub  taxi_start  taxi_1km  taxi_1h  gasolina_1l  \
0                 11.63        0.61      0.22     1.22         1.25   
1                 27.94        2.10      0.42     8.38         1.17   
2                  9.53        0.41      0.29     2.86         0.79   
3                 11.86        0.71      0.24     2.12         1.38   
4                 41.47        2.86      0.90     9.05         1.41   

   volkswagen_coche_nuevo  toyota_coche_nuevo  gastos_basicos_apartamento  \
0                12242.64            22972.31                       57.37   
1                19554.98            20360.82                       64.76   
2                25489.74            27978.73                       80.12   
3                22984.03            18200.60                       97.45   
4                28648.67            26851.06                      176.24   

   precio_movil_1min  internet_men

In [9]:
# Forma del dataframe

df.shape
f' Las filas y las columnas que tiene el dataframe son {df.shape}'


' Las filas y las columnas que tiene el dataframe son (4874, 59)'

In [10]:
# Tamaño/ número de entradas

df.size

287566

In [11]:
# Información general del dataframe

df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4874 entries, 0 to 4873
Data columns (total 59 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   indice                      4874 non-null   int64  
 1   ciudad                      4874 non-null   object 
 2   pais                        4874 non-null   object 
 3   comida_x1_rest              4449 non-null   float64
 4   comida_x2_rest              4432 non-null   float64
 5   menu_mcdonals               4552 non-null   float64
 6   cerveza_nac_rest_0.5cl      4385 non-null   float64
 7   cerveza_imp_rest_0.33cl     4445 non-null   float64
 8   capuccino_rest              4540 non-null   float64
 9   coke_rest_0.33cl            4438 non-null   float64
 10  agua_rest_0.33cl            4571 non-null   float64
 11  leche_1l                    4494 non-null   float64
 12  barra_pan                   4457 non-null   float64
 13  arroz_1kg                   4538 

In [12]:
# Variables numéricas

df.describe().T


count          mean          std      min  \
indice                      4874.0   2436.500000  1407.146936     0.00   
comida_x1_rest              4449.0     10.321643     7.165507     0.45   
comida_x2_rest              4432.0     43.213592    25.365827     3.25   
menu_mcdonals               4552.0      6.875426     2.669017     1.06   
cerveza_nac_rest_0.5cl      4385.0      3.376732     2.080606     0.32   
cerveza_imp_rest_0.33cl     4445.0      3.937287     2.088773     0.31   
capuccino_rest              4540.0      2.742822     1.353734     0.22   
coke_rest_0.33cl            4438.0      1.513053     0.905525     0.11   
agua_rest_0.33cl            4571.0      1.110768     0.749957     0.07   
leche_1l                    4494.0      1.139455     0.513458     0.26   
barra_pan                   4457.0      1.605277     1.079458     0.05   
arroz_1kg                   4538.0      2.182243     1.637878     0.24   
huevos_12ud                 4369.0      2.330382     1.066676     0.10   
queso_1kg                   4393.0      9.448532     4.895459     0.90   
pollo_1kg                   4318.0      7.138745     4.461518     1.17   
ternera_1kg                 4312.0     11.656018     7.203071     1.43   
manzanas_1kg                4509.0      2.747809     1.870736     0.27   
platanos_1kg                4499.0      1.543952     0.826588     0.22   
orange_1kg                  4439.0      2.351935     1.694907     0.20   
tomates_1kg                 4410.0      2.547399     1.810451     0.12   
patatas_1kg                 4444.0      1.568906     1.233074     0.11   
cebollas_1kg                4447.0      1.516267     1.113866     0.11   
lechuga                     4379.0      1.219847     0.729826     0.08   
agua_super_1.5l             4490.0      0.986062     0.679048     0.11   
botella_vino_super          4414.0      8.645111     4.620889     0.36   
cerveza_nac_super_0.5cl     4149.0      1.657585     0.912849     0.25   
cerveza_imp_super_0.33cl    4223.0      2.308425     1.192877     0.35   
tabaco_paq_20ud             4389.0      6.253698     4.431934     0.45   
billete_transporte_pub      3401.0      1.510700     1.316234     0.00   
abono_transporte_pub        2750.0     43.832596    36.519276     0.00   
taxi_start                  4013.0      2.758074     2.209629     0.14   
taxi_1km                    3869.0      1.217227     0.835750     0.10   
taxi_1h                     3506.0     17.124361    15.576644     0.16   
gasolina_1l                 4296.0      1.344767     0.485552     0.03   
volkswagen_coche_nuevo      3839.0  23575.324819  8989.808446     0.00   
toyota_coche_nuevo          3904.0  23674.483617  8704.083758  3166.97   
gastos_basicos_apartamento  4400.0    133.652845    83.129661     6.08   
precio_movil_1min           3702.0      0.133371     0.120924     0.00   
internet_mensual            4534.0    147.808304  7222.936739     1.35   
cuota_gym                   3523.0     36.673032    22.996847     0.00   
alq_pista_tenis             2508.0     15.681116    11.112880     0.00   
entrada_cine                4405.0      8.337280     4.108129     0.68   
guarderia_mensual           3761.0    488.987812   514.976064     4.07   
colegio_anual               3230.0   8684.306341  7879.403518     0.00   
vaqueros_levis              4413.0     52.112579    25.016556     4.07   
vestido_verano              4273.0     34.733440    14.984759     0.00   
deportivas_nike             4388.0     76.389164    33.281138     7.92   
zapatos_cuero_homb          4337.0     85.420108    40.328050     7.57   
apart_1hab_centro           3533.0    710.696156   650.404203    21.69   
apart_1hab_periferia        3465.0    559.472898   529.592777    12.20   
apart_3hab_centro           3418.0   1252.427627  1222.803640    61.20   
apart_3hab_periferia        3385.0    973.603867   926.552107    27.11   
m2_centro                   2681.0   3207.173025  6222.564176   109.53   
m2_periferias               2609.

In [13]:
# Variables categóricas

df.describe(include="object")


ciudad           pais
count       4874           4874
unique      4816            204
top     Victoria  United States
freq           4           1074

In [14]:
# Comprobamos si hay duplicados

df.duplicated().sum()


0

In [15]:
# Porcentaje de nulos de cada columna

df.isnull().sum()/len(df)*100


indice                         0.000000
ciudad                         0.000000
pais                           0.000000
comida_x1_rest                 8.719737
comida_x2_rest                 9.068527
menu_mcdonals                  6.606483
cerveza_nac_rest_0.5cl        10.032827
cerveza_imp_rest_0.33cl        8.801805
capuccino_rest                 6.852688
coke_rest_0.33cl               8.945425
agua_rest_0.33cl               6.216660
leche_1l                       7.796471
barra_pan                      8.555601
arroz_1kg                      6.893722
huevos_12ud                   10.361100
queso_1kg                      9.868691
pollo_1kg                     11.407468
ternera_1kg                   11.530570
manzanas_1kg                   7.488716
platanos_1kg                   7.693886
orange_1kg                     8.924908
tomates_1kg                    9.519902
patatas_1kg                    8.822323
cebollas_1kg                   8.760771
lechuga                       10.155929


In [16]:
df['pais'].unique()

array(['India', 'China', 'Indonesia', 'Philippines', 'South Korea',
       'Thailand', 'Japan', 'Nigeria', 'Pakistan', 'Bangladesh', 'Brazil',
       'Russia', 'Mexico', 'Egypt', 'United States', 'Turkey',
       'Argentina', 'Iran', 'France', 'United Kingdom', 'Vietnam', 'Peru',
       'Angola', 'Malaysia', 'Colombia', 'Tanzania', 'Chile', 'Hong Kong',
       'Saudi Arabia', 'Iraq', 'Spain', 'Sudan', 'Kenya', 'Canada',
       'Singapore', 'Myanmar', 'Australia', 'South Africa', 'Morocco',
       'Afghanistan', 'Jordan', 'Germany', 'Algeria', 'Ethiopia',
       'Kuwait', 'Yemen', 'Ukraine', 'Guatemala', 'Italy', 'Taiwan',
       'Bolivia', 'Ecuador', 'North Korea', 'Madagascar',
       'United Arab Emirates', 'Dominican Republic', 'Uzbekistan',
       'Cameroon', 'Burkina Faso', 'Ghana', 'Azerbaijan', 'Zimbabwe',
       'Cuba', 'Cambodia', 'Somalia', 'Belarus', 'Mali', 'Venezuela',
       'Kazakhstan', 'Austria', 'Malawi', 'Syria', 'Romania', 'Poland',
       'Zambia', 'Belgium', 'Hung

---

**EJERCICIO 2**

Extraer los siguentes estadísiticos del conjunto de datos para varias de las variables disponibles:

- Media

In [17]:
# Media Agua (botella 0.33 en restaurantes)
media_agua1 = df['agua_rest_0.33cl'].mean()
media_agua1

1.1107678844891709

In [18]:
# Media Agua (botella 1.5 en supermercados)
media_agua2 = df['agua_super_1.5l'].mean()
media_agua2

0.9860623608017818

In [19]:
# Media Gasolina (litro)
media_gasolina = df['gasolina_1l'].mean()
media_gasolina

1.3447672253258844

In [20]:
# Media Internet
media_internet = df['internet_mensual'].mean()
media_internet

147.80830392589328

In [21]:
# Media Apartamento (1 habitación en el centro)
media_apart1 = df['apart_1hab_centro'].mean()
media_apart1

710.6961562411549

In [22]:
# Media Apartamento (1 habitación fuera del centro)
media_apart2 = df['apart_1hab_periferia'].mean()
media_apart2

559.4728975468975

In [23]:
# Media Apartamento (3 habitaciónes en el centro)
media_apart3 = df['apart_3hab_centro'].mean()
media_apart3

1252.4276272674078

In [24]:
# Media Apartamento (3 habitaciónes fuera del centro)
media_apart4 = df['apart_3hab_periferia'].mean()
media_apart4

973.6038670605612

In [25]:
# Media Sueldo neto mensual (datos originales de la media ponderada- average)
media_salario = df['media_pond_salario_neto'].mean()
media_salario

1817.033501011853

### Cálculo de estádisticos del agua 

- Desviación con respecto de la media

In [26]:
media_agua2
diferencias = df['agua_super_1.5l'] - media_agua2
desv_med_agua2 = np.abs(diferencias)
desv_med_agua2

0       0.626062
1       0.366062
2       0.566062
3       0.336062
4       0.073938
          ...   
4869         NaN
4870         NaN
4871    0.206062
4872    3.043938
4873         NaN
Name: agua_super_1.5l, Length: 4874, dtype: float64

- Varianza

In [27]:
var_agua2 = df['agua_super_1.5l'] .var()
var_agua2

0.46110572121607496

- Desviación típica

In [28]:
desv_tipica_agua2 = df['agua_super_1.5l'].std()
desv_tipica_agua2

0.6790476575440599

- Moda

In [29]:
moda_agua2 = df['agua_super_1.5l'].mode()
moda_agua2

0    2.0
Name: agua_super_1.5l, dtype: float64

- Mediana

In [30]:
mediana_agua2 = df['agua_super_1.5l'].median()
mediana_agua2

0.75

- Media ponderada

In [31]:
w = np.random.randint(0,100, df.shape[0])
media_pond_agua2 = np.average(df["agua_super_1.5l"], weights=w)
print(media_pond_agua2)
# no es capaz de calcular el average por tener nulos


nan


In [32]:
# Hacemos una copia del df original con la columna de la cual queremos la media ponderada para eliminar 
# los valor nulos y poder hacer el calculo

df_ag = df['agua_super_1.5l'].copy()
df_agua = df_ag.reset_index()
df_agua.head()

index  agua_super_1.5l
0      0             0.36
1      1             0.62
2      2             0.42
3      3             0.65
4      4             1.06

In [33]:
df_agua2 = df_agua.dropna(axis=0, how="any", inplace=False)
df_agua2

index  agua_super_1.5l
0         0             0.36
1         1             0.62
2         2             0.42
3         3             0.65
4         4             1.06
...     ...              ...
4864   4864             4.69
4865   4865             1.61
4867   4867             2.12
4871   4871             0.78
4872   4872             4.03

[4490 rows x 2 columns]

In [34]:
df_agua2.isnull().sum()

index              0
agua_super_1.5l    0
dtype: int64

In [35]:
w = np.random.randint(0,100, df_agua2.shape[0])
media_pond_agua_notnull = np.average(df_agua2["agua_super_1.5l"], weights=w)
print(media_pond_agua_notnull)

0.9849829938093866


- Robustez

In [36]:

robustez = 1 / df.shape[0]
robustez


0.0002051702913418137

In [37]:
n_agua = df_agua.size
rob_agua = 1 / n_agua
rob_agua

0.00010258514567090685

- Coeficiente de variación

In [38]:
coef_var_agua2 = media_agua2 / desv_tipica_agua2
coef_var_agua2

1.4521254139482858

### Cálculo de estadísticos del salario

- Media


In [39]:
# Media Salario neto mensual (datos originales de la media ponderada- average)
media_salario

1817.033501011853

- Desviación con respecto de la media


In [40]:
media_salario
diferencias2 = df['media_pond_salario_neto'] - media_salario
desv_salario = np.abs(diferencias2)
desv_salario

0       1230.683501
1        434.203501
2       1333.843501
3       1398.013501
4        855.196499
           ...     
4869            NaN
4870            NaN
4871            NaN
4872            NaN
4873            NaN
Name: media_pond_salario_neto, Length: 4874, dtype: float64

- Varianza


In [41]:
var_salario = df['media_pond_salario_neto'] .var()
var_salario

2645446.5269653266

- Desviación típica


In [42]:
desv_tipica_salario = df['media_pond_salario_neto'].std()
desv_tipica_salario

1626.4828701727315

- Moda


In [43]:
moda_salario = df['media_pond_salario_neto'].mode()
moda_salario

0    2077.39
Name: media_pond_salario_neto, dtype: float64

- Mediana


In [44]:
mediana_salario = df['media_pond_salario_neto'].median()
mediana_salario

1298.37

- Media ponderada


In [45]:
w = np.random.randint(0,100, df.shape[0])
media_pond_salario = np.average(df["media_pond_salario_neto"], weights=w)
print(media_pond_salario)
# no es capaz de calcular el average por tener nulos

nan


In [46]:
# Hacemos una copia del df original con la columna de la cual queremos la media ponderada para eliminar 
# los valor nulos y poder hacer el calculo

df_sal = df['media_pond_salario_neto'].copy()
df_salario = df_sal.reset_index()
df_salario.head()

index  media_pond_salario_neto
0      0                   586.35
1      1                  1382.83
2      2                   483.19
3      3                   419.02
4      4                  2672.23

In [47]:
df_salario2 = df_salario.dropna(axis=0, how="any", inplace=False)
df_salario2

index  media_pond_salario_neto
0         0                   586.35
1         1                  1382.83
2         2                   483.19
3         3                   419.02
4         4                  2672.23
...     ...                      ...
4863   4863                   612.13
4864   4864                  5867.96
4865   4865                  3435.18
4867   4867                  2471.40
4868   4868                  5588.53

[3459 rows x 2 columns]

In [48]:
w = np.random.randint(0,100, df_salario2.shape[0])
media_pond_salario_notnull = np.average(df_salario2["media_pond_salario_neto"], weights=w)
print(media_pond_salario_notnull)

1806.2087730768776


- Robustez

In [49]:
robustez2 = 1 / df.shape[0]
robustez2

0.0002051702913418137

- Coeficiente de variación

In [50]:
coef_var_salario = media_salario / desv_tipica_salario
coef_var_salario

1.1171550185578563

---

**EJERCICIO 3**

Interpretación de los resultados.

- Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.


Trabajamos con datos sobre el **coste de la vida** en diferentes ciudades del mundo, primero exploramos el archivo observando que los datos se basan en **4816 ciudades** de **204 países** 
diferentes, con **54 artículos/variables** y sus correspondientes precios, con un **total de 287566 entradas**.

Elegimos dos variables para obtener los diferentes estadísticos de las mismas, la columna **'agua_super_1,5l'** entendido como bien básico y la columna **'media_pond_salario_neto'**  
entendido como medio vital necesario, en la segunda columna las cifras del salario no vienen dados de la media sino de la media ponderada por lo que tiene mayor peso.

- Obtenemos de la columna de agua embotellada de 1.5 litros comprada en el supermercado las siguientes estadísticas:

    - Media: 0.9860623608017818
    - Varianza: 0.46110572121607496
    - Desviación típica: 0.6790476575440599
    - Moda: 2.0
    - Mediana: 0.75
    - Media ponderada: 0.9924516424916594
    - Robustez: 0.0002051702913418137
    - Coeficiente de variación: 1.4521254139482858



- Obtenemos de la columna media ponderada de salario neto las siguientes estadísticas:

    - Media: 1817.033501011853
    - Varianza: 2645446.5269653266
    - Desviación típica: 1626.4828701727315
    - Moda: 2077.39
    - Mediana: 1298.37
    - Media ponderada: 1807.98536619965
    - Robustez: 0.0002051702913418137
    - Coeficiente de variación: 1.1171550185578563




Con los datos del salario, teniendo en cuenta que los datos obtenidos incialmente venian dados de una media ponderada con su peso y robustez, podemos ver que la media y la media ponderada obtenida de los cálculos estadísticos no son tan diferentes aunque la media ponderada se ajusta hacia abajo si tenemos en cuenta la cifra tan baja de los sueldos en los países donde hay más pobreza.

Por otra parte, la moda o cantidad más frecuente ronda los 2077.39, estando por encima de la media, apuntando a que gran cantidad de países tienen sueldos medios-altos, viendo tambíen que la desviación típica es bastante alta, debido a la gran diferencia de sueldos en los países mas y menos ricos.

Con la elevada cantidad de la varianza podemos intuir que hay sueldos muy altos y que con el coeficiente de variación bastante homogéneo esa variación es solo para una pequeña parte de la población mundial.

---

In [51]:
# Guardamos el archivo con las modificaciones

df.to_csv('cost_of_living2.csv')